### PR TITLE
[Object Detection] Fix SmoothL1 Loss Calculation

### LINKED ISSUE #920: smooth l1 loss does not match formula
Here's our implementation:
https://github.com/keras-team/keras-cv/blame/master/keras_cv/losses/smooth_l1.py#L39-L42

Here's a definition used by PyTorch:
https://pytorch.org/docs/stable/generated/torch.nn.SmoothL1Loss.html

A beta denominator is missing. It still works fine if the cutoff is 1, but anything else would be different.
We should also add more unit test to it while fixing it

### PR DESCRIPTION
**Pull request by DavidLandup0**
# What does this PR do?

- Updates the SmoothF1 Loss to correctly account for the `beta`.
- Renamed `f1_cutoff` to `beta` for consistency with paper terminology and refactored `f1_cutoff` usage.
- Added numerical tests for the `beta` argument.

Fixes #920

## Who can review?

Tagging original authors (@LukeWood @fchollet) and @tanzhenyu for opening the issue.

### CODE REVIEW

**Comment by LukeWood** [comment]
Hey @DavidLandup0 Thanks for the PR!  Whats the motivation for the change?  

We picked the name `l1_cutoff` as it is more clear from a readability perspective than `beta`.  many readers can figure out what an l1 cutoff is without looking at the formula, beta you need to look at the formula.

File: keras_cv/losses/smooth_l1.py [hunk 1]
```diff
@@ -19,33 +19,33 @@
...
     def call(self, y_true, y_pred):
         difference = y_true - y_pred
         absolute_difference = tf.abs(difference)
         squared_difference = difference**2
         loss = tf.where(
-            absolute_difference < self.l1_cutoff,
-            0.5 * squared_difference,
-            absolute_difference - 0.5,
+            absolute_difference < self.beta,
+            (0.5 * squared_difference) / self.beta,
```
Comment (LukeWood) [inline keras_cv/losses/smooth_l1.py:41]: ah I see that this actually changes the numerics

Comment (DavidLandup0) [reply keras_cv/losses/smooth_l1.py:41]: Yup - the original name was a clear use of the argument, since it was just a cutoff. It works perfectly the same as the change when the cutoff is 1.0 (default), but as soon as it changes, the loss was technically incorrect.
We can still keep the cutoff name for backwards compatability if you think that's a good idea, but I think we'd benefit more from having it change early on to a slightly more accurate term?

Comment (DavidLandup0) [reply keras_cv/losses/smooth_l1.py:41]: Reference formulas:

- https://pytorch.org/docs/stable/generated/torch.nn.SmoothL1Loss.html
- https://paperswithcode.com/method/self-adjusting-smooth-l1-loss

Comment (DavidLandup0) [reply keras_cv/losses/smooth_l1.py:41]: We should actually lose the parentheses in the second statement. Updating in a moment

Comment (DavidLandup0) [reply keras_cv/losses/smooth_l1.py:44]: Updated and updated docs to make the effect of `beta` clearer!

Comment (DavidLandup0) [reply keras_cv/losses/smooth_l1.py:47]: Pinging since I think that the fix for the loss func is relevant for the 0.4 release @LukeWood 
Not blocking by any means, but worth having if we're releasing OD/SemSeg

**Comment by DavidLandup0** [comment]
Pinging again since this is a small change that might impact some of the workflows currently underway
@tanzhenyu

**Comment by LukeWood** [comment]
Lets keep the naming convention.  In keras we prefer clarity over matching publications.  This is done as the ML landscape is ever-changing, and its impossible to keep track of every specific naming convention.   Better to give an indicative name, `cutoff` than `beta`
